stdlib/src/utils/static_tuple.mojo

```diff
@@ -110,7 +110,8 @@ fn _static_tuple_construction_checks[size: Int]():
     Parameters:
       size: The number of elements.
     """
-    constrained[size > 0, "number of elements in `StaticTuple` must be > 0"]()
+    pass
+    # constrained[size > 0, "number of elements in `StaticTuple` must be > 0"]()
```

Comment:
This is going to take a bit of wrangling. We were just talking about this constraint internally; its needed for size/alignment issues, but as features like this PR exemplify, that restriction might be handled better on a case-by-case basis.